pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -191,42 +193,84 @@ func (c *defaultMaintenanceControl) Maintain(shootObj *gardencorev1beta1.Shoot,
+func MaintainKubernetesVersion(shoot *gardencorev1beta1.Shoot, profile *gardencorev1beta1.CloudProfile, logger *logrus.Entry) (*string, *string, error) {
+	shouldBeUpdated, reason, isExpired, err := shouldKubernetesVersionBeUpdated(shoot, profile)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if shouldBeUpdated {
-		newerPatchVersionFound, latestPatchVersion, err := gardencorev1beta1helper.DetermineLatestKubernetesPatchVersion(profile, shoot.Spec.Kubernetes.Version)
+		// get latest version that qualifies for a patch update
+		newerPatchVersionFound, latestPatchVersion, err := gardencorev1beta1helper.GetKubernetesVersionForPatchUpdate(profile, shoot.Spec.Kubernetes.Version)
 		if err != nil {
-			return nil, fmt.Errorf("failure while determining the latest Kubernetes patch version in the CloudProfile: %s", err.Error())
+			return nil, nil, fmt.Errorf("failure while determining the latest Kubernetes patch version in the CloudProfile: %s", err.Error())
 		}
 		if newerPatchVersionFound {
-			return &latestPatchVersion, nil
+			msg := fmt.Sprintf("Kubernetes version '%s' to version '%s'. This is an increase in the patch level. Reason: %s", shoot.Spec.Kubernetes.Version, latestPatchVersion, *reason)
+			logger.Debugf("[SHOOT MAINTENANCE] Updating %s", msg)
+			return &latestPatchVersion, &msg, nil
+		}
+		// no newer patch version found & is expired -> forcefully update to latest patch of next minor version
+		if isExpired {
+			// get latest version that qualifies for a minor update
+			newMinorAvailable, latestPatchVersionNewMinor, err := gardencorev1beta1helper.GetKubernetesVersionForMinorUpdate(profile, shoot.Spec.Kubernetes.Version)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failure while determining newer Kubernetes minor version in the CloudProfile: %s", err.Error())
+			}
+			// cannot update as there is no consecutive minor version available (e.g shoot is on 1.13.X, but there is only 1.15.X, available and not 1.14.X)
+			if !newMinorAvailable {
+				return nil, nil, fmt.Errorf("cannot update expired Kubernetes version '%s'. No suitable version found in CloudProfile", shoot.Spec.Kubernetes.Version)
+			}
+
+			msg := fmt.Sprintf("Kubernetes version '%s' to version '%s'. This is an increase in the minor level. Reason: %s", shoot.Spec.Kubernetes.Version, latestPatchVersionNewMinor, *reason)
+			logger.Debugf("[SHOOT MAINTENANCE] Updating %s", msg)
+			return &latestPatchVersionNewMinor, &msg, nil
 		}
 	}
-	return nil, nil
+	return nil, nil, nil
 }
 
-func shouldKubernetesVersionBeUpdated(shoot *gardencorev1beta1.Shoot, profile *gardencorev1beta1.CloudProfile) (bool, error) {
-	versionExistsInCloudProfile, offeredVersion, err := gardencorev1beta1helper.KubernetesVersionExistsInCloudProfile(profile, shoot.Spec.Kubernetes.Version)
+func shouldKubernetesVersionBeUpdated(shoot *gardencorev1beta1.Shoot, profile *gardencorev1beta1.CloudProfile) (bool, *string, bool, error) {
```

Comment:
Can you please give the return parameters names, if you have multiple `bool`s, or at least add a doc string to make it easier to understand?